server/apps/immich/src/api-v1/album/album-repository.ts

```diff
@@ -0,0 +1,228 @@
+import { AlbumEntity } from '@app/database/entities/album.entity';
+import { AssetAlbumEntity } from '@app/database/entities/asset-album.entity';
+import { UserAlbumEntity } from '@app/database/entities/user-album.entity';
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { getConnection, Repository, SelectQueryBuilder } from 'typeorm';
+import { AddAssetsDto } from './dto/add-assets.dto';
+import { AddUsersDto } from './dto/add-users.dto';
+import { CreateAlbumDto } from './dto/create-album.dto';
+import { GetAlbumsDto } from './dto/get-albums.dto';
+import { RemoveAssetsDto } from './dto/remove-assets.dto';
+import { UpdateAlbumDto } from './dto/update-album.dto';
+
```

Comment:
I believe this pattern is from .NET correct? 

Why do you think it is necessary to go down another layer of obstraction vs the original flow that incorporate the repository/query directly into the `service`? I am new to design the API system so I am here to learn :)

Reply:
It's all about decoupling "layers", which among other things makes testing easier. These decoupling concepts are part of architectures like Onion, Clean, Hexagonal.
Immich is mostly Onion architecture